File: languages/java/src/provider/SummaryAnalysis.java
package provider;

import model.TableAnalysis;
import model.UserInfo;
import model.exception.InvalidParameterException;

import java.util.List;

/**
 * Classe para analise de dados, obtem os valores maximo e minimo dos creditos dos usuarios
 * @see model.TableAnalysis
 */
public class SummaryAnalysis implements TableAnalysis<double[]> {

    /**
     * Realiza a analise maxima e minima dos dados
     * @param userInfoList Lista de dados a ser analisada
     * @return Valores maximo e minimo dos creditos
     * @see TableAnalysis
     */
    @Override
    public double[] analysis(List<UserInfo> userInfoList) {
        if(userInfoList == null || userInfoList.isEmpty())
            throw new InvalidParameterException("'userInfoList' é null ou vazio");

        double max = Double.MIN_VALUE;
        double min = Double.MAX_VALUE;
        double sum = 0;

        for(UserInfo userInfo : userInfoList){
            double credit = userInfo.getCredit();
            sum += credit;
            if(max < credit) max = credit;
            if(min > credit) min = credit;
        }

        double mean = sum/userInfoList.size();
        return new double[]{min, max, mean};
    }
}

File: languages/dotnet/src/provider/SummaryAnalysis.cs
using experimento.src.model;
using experimento.src.model.exception;
using System.Collections.Generic;

namespace experimento.src.provider {

    /// <summary>
    ///  Classe para analise de dados, obtem a media dos valores dos creditos dos usuarios
    /// </summary>
    public class SummaryAnalysis : TableAnalysis<double[]> {
        
        /// <summary>
        /// Realiza a analise de media dos dados
        /// </summary>
        /// <param name="list">Lista de dados a ser analisada</param>
        /// <returns>Media dos valores dos creditos</returns>
        public double[] analysis(List<UserInfo> list) {
            if(list == null || (list.Count == 0))
                throw new InvalidParameterException("'userInfoList' é null ou vazio");

            double sum = 0;
            double min = double.MaxValue;
            double max = double.MinValue;
            
            foreach(UserInfo userInfo in list) {
                sum += userInfo.credit;
                if(min > userInfo.credit) min = userInfo.credit;
                if(max > userInfo.credit) max = userInfo.credit;
            }

            double mean = sum/list.Count;
            double[] range = {min, max, mean};
            return range;
        }
    }
}